### ISSUE TITLE
[HOLD Safari 17.4][$1000] [mWeb/Safari] Fix safari unfocused selection prop

### ISSUE THREAD

**Issue by thienlnam** [Expensify/App#26239]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Break down in numbered steps
1. We have a bug that is specific to mWeb/Safari that causes us to continue making safari specific updates in our code

https://github.com/Expensify/App/blob/bb2386beacf38d83ec438291bd4ca3684e9bf0ad/src/pages/home/report/ReportActionItemMessageEdit.js#L120-L122

https://github.com/Expensify/App/pull/24369/commits/f74dbb1afe2cbfe35fb167c8d87ccdfd18883449

2. To reduce the amount of platform specific code in our repo - let's aim to get this fixed upstream. Creating this issue to track for when we can do this

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b2f48b6c5a3887bd</li>
        <li>Upwork Job ID: 1700115117826949120</li>
        <li>Last Price Increase: 2024-06-05</li>
<li>Automatic offers: </li>
<ul>
<li>getusha | Contributor | 102518816</li>
</ul></li>
    </ul>
</details>

**Comment by thienlnam** [comment]
Adding monthly as an upstream fix might take a while

**Comment by michaelhaxhiu** [comment]
As mentioned by @thienlnam, this will get fixed upstream.

>To reduce the amount of platform specific code in our repo - let's aim to get this fixed upstream. Creating this issue to track for when we can do this

We need a contributor to make the corresponding GH in that repository and then we'll collectively push to get this across the finish line. 👍

**Comment by michaelhaxhiu** [comment]
@ntdiary You want to submit the GH on the upstream repo to drive this through ⛳ ?

**Comment by michaelhaxhiu** [comment]
Thanks @ntdiary :) 

I think this is definitely worth finishing off. @thienlnam do you agree with the bounty price on this btw (e.g. it's fair for the effort and time involved all things considered)? 

I'm assigning you since you made this GH and there's no engineer.

**Comment by thienlnam** [comment]
Issues like this will take longer just due to the nature of having to get code accepted into a different repo that we don't really have control over. That being said, it would be nice to get this done as we have a few hacks added in our codebase to make a workaround for this.

Let's increase this to $1000 and just see where it goes

**Comment by thienlnam** [comment]
This is an upstream fix and will take time

**Comment by alexpensify** [comment]
@ntdiary any update on the PR refactor?

**Comment by alexpensify** [comment]
Awesome, keep us posted if you need a hand or review.

**Comment by alexpensify** [comment]
Thank you for the update, keep us posted on how it's going.

**Comment by alexpensify** [comment]
@ntdiary - are you getting responses? If no, do you need me to reach out to the repo maintainers? I can do so early next week.

**Comment by alexpensify** [comment]
Daily Update: Looks like there is movement in the PR.

**Comment by alexpensify** [comment]
Let's go! Thanks for the update.

**Comment by thienlnam** [comment]
Wooo! Congratulations 🎉 Great to hear

This will definitely help us clean up a lot of our safari specific handling. Do you know when the new versions release for safari?

**Comment by alexpensify** [comment]
@thienlnam - any feedback for the next steps here?

**Comment by thienlnam** [comment]
I believe we just need to keep this open to verify the fix once the next version of Safari is deployed - which may take a couple months

**Comment by alexpensify** [comment]
@thienlnam should we spin out another GH specific for the Safari update and pay for the work completed so far or wait for Safari? Thanks for clarifying!

**Comment by thienlnam** [comment]
Good question, we usually only pay once we verify the fix is confirmed - though the issue with this specific issue is we can't verify the fix until the Safari pushes those changes so I believe we'd have to wait for that here

**Comment by alexpensify** [comment]
Sounds like a plan! We will check in a month then.

**Comment by alexpensify** [comment]
Safari 17.3 has not been released yet, setting a reminder to check in 1 month.

**Comment by alexpensify** [comment]
@thienlnam - I believe that the Safari version we are waiting on is ready. Can you confirm if we can move this one from being on hold? Thanks!

**Comment by thienlnam** [comment]
I'm not seeing the version we're waiting for yet - this seems to be the latest Version 17.2.1

**Comment by alexpensify** [comment]
Done and thank you for flagging the version to watch out for.

**Comment by alexpensify** [comment]
Monthly Update: We are at 17.3.1. Setting a reminder to check in next month.

**Comment by alexpensify** [comment]
@ntdiary - I believe that we are at Version 17.4.1 now, can we confirm if we can take this one off hold? Thanks!

**Comment by alexpensify** [comment]
@ntdiary - We need proposals now, right?

**Comment by thienlnam** [comment]
We're good to pay out now - this should be resolved with the latest version!

**Comment by thienlnam** [comment]
Yeah that would be great if we could clean up the codebase, please do!

**Comment by tomekzaw** [comment]
> do you think we can move forward with the above react-native-live-markdown change?

I believe it's a question for @Skalakid or @BartoszGrajdek as they have better knowledge of the web implementation of Live Markdown.

**Comment by alexpensify** [comment]
Update: We are waiting for this PR: https://github.com/Expensify/App/pull/40128. After that, @ntdiary is there any other action here?

**Comment by alexpensify** [comment]
Thank you for that update, it looks like the other PR is moving along too.

**Comment by alexpensify** [comment]
The PR we were waiting on is in Staging now.

**Comment by alexpensify** [comment]
@ntdiary - The PR we were waiting on went into production. We are ready for https://github.com/Expensify/App/pull/40295. Thanks!

**Comment by alexpensify** [comment]
Sounds like a plan! Thanks for the update.

**Comment by alexpensify** [comment]
Thank you for the update @ntdiary!

**Comment by alexpensify** [comment]
Weekly Update: I checked the PR and we are moving forward. Thank you @ntdiary!

**Comment by alexpensify** [comment]
## Heads up, I will be offline until Tuesday, May 28, 2024, and will not actively watch over this GitHub during that period.

If anything urgent is needed here, please ask for help in the #expensify-open-source Slack Room-- thanks!

**Comment by alexpensify** [comment]
Update: We are waiting for this PR to go into production.

**Comment by alexpensify** [comment]
@ntdiary - noted, I'll still work on the summary later today but not issue a payment in Upwork. When you are ready, you can input the Request and point back to the summary.

**Comment by alexpensify** [comment]
Payouts due: 2024-05-29

- [ ] Contributor: $500 @ntdiary 
- [ ] Contributor+: $500 @getusha 

Upwork job is [here]().

**Comment by alexpensify** [comment]
@getusha can you access this offer here: 

https://www.upwork.com/nx/wm/offer/102518816

If yes, please accept and then I'll complete the payment process. Thanks!

**Comment by alexpensify** [comment]
@getusha any update if you can access that Upwork link to approve? Thanks!

**Comment by getusha** [comment]
> @getusha any update if you can access that Upwork link to approve? Thanks!

Apologies for the delay, i am also working on migrating payments to ND, could we hold it? thanks!

**Comment by alexpensify** [comment]
@getusha - sounds like a plan. Keep me posted when you are set up. For now, I'm going to withdraw the job in Upwork to close the job.

**Comment by alexpensify** [comment]
@ntdiary - Here is the update from April 4 regarding the changes to the job prices: 

https://expensify.slack.com/archives/C01GTK53T8Q/p1712256409152489

**Comment by alexpensify** [comment]
I've reviewed the summary again and realized that I based it off the PR creation date. My mistake, the updated summary is here: 

https://github.com/Expensify/App/issues/26239#issuecomment-2138377121

**Comment by JmillsExpensify** [comment]
$1,000 approved for @ntdiary

**Comment by alexpensify** [comment]
@getusha, are you set up for payments, so we can close out this issue? Thanks!

**Comment by getusha** [comment]
> @getusha, are you set up for payments, so we can close out this issue? Thanks!

@alexpensify not yet, i received an invitation last week. I think the process is yet to begin.

**Comment by alexpensify** [comment]
Thanks for the update!

**Comment by getusha** [comment]
@alexpensify since i was assigned before 2024-05-25 can we process the payment via upwork?

**Comment by alexpensify** [comment]
@getusha ok, I sent over a new offer in Upwork. Please accept here: https://www.upwork.com/jobs/~019cb418da52a59d59

I can complete the next steps after. Thanks!

**Comment by getusha** [comment]
@alexpensify, Accepted, thank you!

**Comment by alexpensify** [comment]
All set here, I've completed the payment process to @getusha. Everyone has been paid accordingly: https://github.com/Expensify/App/issues/26239#issuecomment-2138377121.
